octopus-agent/src/main/java/octopus/teamcity/agent/InterruptableBuildProcess.java

```diff
@@ -15,25 +15,48 @@
 
 package octopus.teamcity.agent;
 
+import static octopus.teamcity.agent.logging.SdkLogAppender.SDK_APPENDER_NAME;
+import static octopus.teamcity.agent.logging.SdkLogAppender.isVerboseLogging;
+
+import com.octopus.sdk.logging.SdkLogAppenderHelper;
+
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
 import jetbrains.buildServer.RunBuildException;
 import jetbrains.buildServer.agent.BuildFinishedStatus;
 import jetbrains.buildServer.agent.BuildProcess;
+import jetbrains.buildServer.agent.BuildRunnerContext;
+import octopus.teamcity.agent.logging.SdkLogAppender;
 
 public abstract class InterruptableBuildProcess implements BuildProcess {
 
   private volatile boolean interrupted = false;
   private final CompletableFuture<BuildFinishedStatus> uploadFinishedFuture =
       new CompletableFuture<>();
 
-  public InterruptableBuildProcess() {}
+  private final BuildRunnerContext context;
+
+  public InterruptableBuildProcess(BuildRunnerContext context) {
+    this.context = context;
+  }
 
   protected void complete(final BuildFinishedStatus status) {
     uploadFinishedFuture.complete(status);
   }
 
+  public abstract void doStart() throws RunBuildException;
+
+  @Override
+  public void start() throws RunBuildException {
+    try (SdkLogAppenderHelper ignored =
+        SdkLogAppenderHelper.registerLogAppender(
+            SdkLogAppender.createAppender(SDK_APPENDER_NAME, context.getBuild().getBuildLogger()),
```

Comment:
I know we're the only user of this - but the SDK_APPENDER_NAME should be hidden in the SDK somehow.
Its a bit heavy, but we've got time - maybe we can try putting a lambda creating FunctionalInterface

SDK offers:
@FunctionaInterface
class AppenderCreator {
   Appender create();
}

SdkLogAppenderHelper(AppenderCreator creator) {
  Appender = creator.create(OCTOPUS_APPENDER_NAME);
...
}



TC uses:
SdkLogAppenderHelper.registerLogAppender(sdk_name -> new LogAppender(sdk_name, BuildProgressLogger));

Means that the OCTOPUS_APPENDER_NAME doesn't leak out of the SDK